internal/static/app_test.go

```diff
@@ -0,0 +1,149 @@
+package static_test
+
+import (
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/configkit/internal/entity"
+	. "github.com/dogmatiq/configkit/internal/static"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"golang.org/x/tools/go/packages"
+)
+
+var _ = Describe("func FromPackages()", func() {
+	When("a package contains a single Dogma application", func() {
+		It("returns the configuration for the application detected", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/apps/single-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+
+			Expect(apps).To(Equal(
+				[]configkit.Application{
+					&entity.Application{
+						IdentityValue: configkit.Identity{
+							Name: "<app>",
+							Key:  "8a6baab1-ee64-402e-a081-e43f4bebc243",
+						},
+						TypeNameValue: "github.com/dogmatiq/configkit/internal/static/testdata/apps/single-app.App",
+						MessageNamesValue: configkit.EntityMessageNames{
+							Produced: nil,
+							Consumed: nil,
+						},
+						HandlersValue: nil,
+					},
+				},
+			))
+		})
+	})
+
+	When("a package contains multiple Dogma applications", func() {
+		It("returns the configuration for all applications detected", func() {
```

Comment:
Do we need a test that ensures it properly loads multiple apps defined in the exact same package?

Reply:
Probably not at all, removed in https://github.com/dogmatiq/configkit/pull/118/commits/736ca39b529d2c2ba600a7f22f592ac99b4dc276.

Reply:
Sorry, we misunderstood each other - I meant, do we need an additional test that ensures we can load multiple app types from one single package. 

The test you removed was testing loading multiple apps across multiple packages. I wasn't suggesting removing it.

Reply:
Oh, I see, I will restore it and get the new test in, sorry 😐!

Reply:
Ok https://github.com/dogmatiq/configkit/pull/118/commits/7653cc8b54140c1e38006b715e0a936824a1ddd4 and https://github.com/dogmatiq/configkit/pull/118/commits/8f456c8817e6349437693c796a4b9bc384e36270 should take care of it.